src/main/java/com/linkedin/kmf/services/ProduceService.java

```diff
@@ -150,7 +150,7 @@ private void initializeProducer() throws Exception {
   public void start() {
     if (_running.compareAndSet(false, true)) {
       initializeStateForPartitions();
-      _handleNewPartitionsExecutor.scheduleWithFixedDelay(new NewPartitionHandler(), 40_000, 40_000, TimeUnit.MILLISECONDS);
+      _handleNewPartitionsExecutor.scheduleWithFixedDelay(new NewPartitionHandler(), 1000, 30000, TimeUnit.MILLISECONDS);
```

Comment:
I think it is OK to just use 1 second delay by default. If we add this check, the only benefit is that if the topic is already created, the first execution of this task can be later. But this saving doesn't seem to be worth the the extra logic (or lines of code). What do you think?